cocos/misc/lodgroup-component.ts

```diff
@@ -0,0 +1,616 @@
+            return;
+        }
+        this._LODs[index] = lod;
+    }
+
+    /**
+     * @en Recalculate the bounding box, and the interface will recalculate the localBoundaryCenter and objectSize
+     * @zh 重新计算包围盒，该接口会更新 localBoundaryCenter 和 objectSize
+     */
+    public recalculateBounds () {
+        LODGroupEditorUtility.recalculateBounds(this);
+    }
+
+    /**
+     * @en reset current objectSize to 1, and recalculate screenUsagePercentage.
+     * @zh 重置 objectSize 的大小为1，该接口会重新计算 screenUsagePercentage
+     */
+    public resetObjectSize () {
+        LODGroupEditorUtility.resetObjectSize(this);
+    }
+
+    onLoad () {
+        this._lodGroup.node = this.node;
+        if (!this._eventRegistered) {
+            this.node.on(NodeEventType.COMPONENT_REMOVED, this._onRemove, this);
+            this._eventRegistered = true;
+        }
+        // generate default lod for lodGroup
+        if (this.lodCount < 1) {
+            const size = DEFAULT_SCREEN_OCCUPATION.length;
+            for (let i = 0; i < size; i++) {
+                this.insertLOD(i, DEFAULT_SCREEN_OCCUPATION[i]);
+            }
+        }
+    }
+
+    _onRemove (comp: Component) {
+        if (comp === this) {
+            this.onDisable();
+        }
+    }
+
+    // Redo, Undo, Prefab restore, etc.
+    onRestore () {
+        if (this.enabledInHierarchy) {
+            this._attachToScene();
+        }
+    }
+
+    onEnable () {
+        this._attachToScene();
+        //   LODGroupEditorUtility.recalculateBounds(this);
+
+        // cache lod for scene
+        if (this.lodCount > 0 && this._lodGroup.lodCount < 1) {
+            this._LODs.forEach((lod: LOD, index) => {
+                lod.lodData.screenUsagePercentage = lod.screenUsagePercentage;
+                const renderers = lod.renderers;
+                let modelCount = 0;
+                if (renderers !== null && renderers.length > 0) {
+                    for (let i = 0; i < renderers.length; i++) {
+                        const lodInstance = lod.lodData;
+                        const renderer = renderers[i];
+                        if (lodInstance && renderer && renderer.model) {
+                            lodInstance.models[modelCount] = renderer.model;
+                            modelCount += 1;
+                        }
+                    }
+                }
+                this._lodGroup.LODs[index] = lod.lodData;
+            });
+        }
+    }
+
+    onDisable () {
+        this._detachFromScene();
+    }
+
+    private _attachToScene () {
+        if (!this.node.scene) { return; }
+
+        const renderScene = this._getRenderScene();
+        if (this._lodGroup.scene) {
+            this._detachFromScene();
+        }
+        renderScene.addLODGroup(this._lodGroup);
+    }
+
+    private _detachFromScene () {
+        if (this._lodGroup.scene) { this._lodGroup.scene.removeLODGroup(this._lodGroup); }
+    }
+}
+
+export class LODGroupEditorUtility {
```

Comment:
It is better to implement in a new ts file. And current ts file can not depend on the new ts file.